community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/NeoStoreUtil.java

```diff
@@ -49,47 +46,57 @@
+            }
+        }
+        catch ( IOException e )
+        {
+            return false;
         }
-        System.out.println( new NeoStoreUtil( new File( args[0] ) ) );
-    }
 
-    public static boolean neoStoreExists( FileSystemAbstraction fs, File storeDir )
-    {
-        return fs.fileExists( neoStoreFile( storeDir, StoreFileType.STORE ) );
+        return true;
     }
 
-    public NeoStoreUtil( File storeDir )
-    {
-        this( storeDir, new DefaultFileSystemAbstraction() );
-    }
-
-    public NeoStoreUtil( File storeDir, FileSystemAbstraction fs )
+    public NeoStoreUtil( File storeDir, PageCache pageCache )
     {
         File neoStoreFile = neoStoreFile( storeDir, StoreFileType.STORE );
-        String currentTypeDescriptorAndVersion = NeoStore.buildTypeDescriptorAndVersion( TYPE_DESCRIPTOR );
-        boolean storeHasTrailer = hasTrailer( neoStoreFile, fs, currentTypeDescriptorAndVersion );
-        try ( StoreChannel channel = fs.open( neoStoreFile, "r" ) )
+        try ( PagedFile pagedFile = pageCache.map( neoStoreFile, pageCache.pageSize() ) )
         {
-            int contentSize = (int) channel.size();
-            if ( storeHasTrailer )
+            if ( pagedFile.getLastPageId() != -1 )
             {
-                int trailerSize = encode( currentTypeDescriptorAndVersion ).length;
-                contentSize -= trailerSize;
-            }
-            int records = contentSize/RECORD_SIZE;
-            ByteBuffer buf = ByteBuffer.allocate( records * RECORD_SIZE );
-            channel.read( buf );
-            buf.flip();
-
-            for ( int i = 0; buf.remaining() >= RECORD_SIZE && i < Position.values().length; i++ )
-            {
-                values.put( Position.values()[i], nextRecord( buf ) );
+                try ( PageCursor cursor = pagedFile.io( 0, PagedFile.PF_SHARED_LOCK ) )
+                {
+                    if ( cursor.next() )
+                    {
+                        byte[] data;
+                        do
+                        {
+                            data = new byte[Position.values().length * RECORD_SIZE];
+                            cursor.getBytes( data );
+                        }
+                        while ( cursor.shouldRetry() );
+                        ByteBuffer buf = ByteBuffer.wrap( data );
+                        int pos = 0;
+                        while ( pos < Position.values().length && buf.remaining() >= RECORD_SIZE )
+                        {
+                            if ( buf.get() == Record.IN_USE.byteValue() )
+                            {
+                                values.put( Position.values()[pos], buf.getLong() );
+                            }
+                            else
+                            {
+                                buf.getLong();
+                            }
+                            pos++;
+                        }
+                    }
+                }
```

Comment:
We do not need to know if there is a trailer or not, since we now use the Record.IN_USE byte to find records.